### PR TITLE
feat(parents) : #MA-1011 parent front page now shows how the leaves are calculated (days, half days, ...).

### PR DESCRIPTION
**Pull request by FlorentMr**
## Describe your changes
Parent front page now shows how the leaves are calculated (days, half days, ...). The feature was half there but was removed with an evolution, did some changes to put it back to work.

## Checklist tests
Se connecter en tant que parent dans présences -> Vérifier que en dessous du chiffre des absences, on ait le type de comptage affiché en petit (jour / demi jour / ...), à tester en changeant ce type tout en bas des settings Présences dans Viesco. 

## Issue ticket number and link
[MA-1011](https://jira.support-ent.fr/browse/MA-1011)

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [x] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: presences/src/main/java/fr/openent/presences/service/impl/DefaultEventStudentService.java [hunk 1]
```diff
@@ -39,6 +40,7 @@ public class DefaultEventStudentService implements EventStudentService {
 
     private final EventService eventService;
     private final ReasonService reasonService = new DefaultReasonService();
+    private final SettingsService settingsService = new DefaultSettingsService();
```
Comment (BastienBenard) [inline presences/src/main/java/fr/openent/presences/service/impl/DefaultEventStudentService.java]: Il y a pas un ServiceFactory de disponible ? Ce serait mis de l'utiliser pour évité la création d'instance d'object

File: presences/src/main/java/fr/openent/presences/service/impl/DefaultEventStudentService.java [hunk 2]
```diff
@@ -123,14 +125,16 @@ private Future<JsonObject> setSettingsAndReasons(String structureId, List<String
 
         // Get reasons types to map it thanks reason_id on absences justified events results
         Future<Map<Integer, JsonObject>> reasonFuture = getReasons(structureId);
+        // Get presence settings to get recovery method (for absences events configuration retrieving)
+        Future<JsonObject> presencesSettingsFuture = settingsService.retrieve(structureId);
```
Comment (BastienBenard) [inline presences/src/main/java/fr/openent/presences/service/impl/DefaultEventStudentService.java]: Ce serait cool de polymorpher la fonction pour utiliser un model. A confirmer par @flmartineau

**Review by flmartineau — APPROVED** [review]

File: gradle.properties [hunk 3]
```diff
@@ -2,7 +2,7 @@
 modowner=fr.openent
 # Your module name
 # Your module version
-version=1.4-SNAPSHOT
+version=1.5-SNAPSHOT
```
Comment (flmartineau) [inline gradle.properties]: met a jour ta branche dev149&151

Comment (FlorentMr) [reply gradle.properties]: je vais sûrement cherry pick les commits vers dev, mais je le fais yep (edit : je l'avais déjà fait, juste mal fait le rebasing, mb)

File: presences/src/main/java/fr/openent/presences/service/SettingsService.java [hunk 4]
```diff
@@ -1,14 +1,23 @@
...
+import fr.openent.presences.model.Settings;
 import fr.wseduc.webutils.Either;
 import io.vertx.core.*;
+import io.vertx.core.json.JsonArray;
 import io.vertx.core.json.JsonObject;
 
+import java.util.Map;
+
 public interface SettingsService {
     void retrieve(String structureId, Handler<Either<String, JsonObject>> handler);
```
Comment (BastienBenard) [inline presences/src/main/java/fr/openent/presences/service/SettingsService.java:22]: java docs pour le deprecated + annotation

File: presences/src/main/java/fr/openent/presences/service/SettingsService.java [hunk 5]
```diff
@@ -1,14 +1,23 @@
...
 
+    /**
+     * @deprecated Replaced by {@link #retrieveSettings(String)}
+     */
     Future<JsonObject> retrieve(String structureId);
```
Comment (BastienBenard) [inline presences/src/main/java/fr/openent/presences/service/SettingsService.java:31]: annotation deprecated

**Review by BastienBenard — APPROVED** [review]

**Pull request merged**